src/redux/data.ts

```diff
@@ -403,42 +866,87 @@ const checkForConfirmedSavingsActions = transactionsData => dispatch => {
   }
 };
 
-const checkForUpdatedNonce = transactionData => dispatch => {
+/**
+ * Checks to see if a network's nonce should be incremented for an acount
+ * based on incoming transaction data, and if so, updates state.
+ *
+ * @param transactionData Incoming transaction data.
+ */
+const checkForUpdatedNonce = (transactionData: ZerionTransaction[]) => (
+  dispatch: ThunkDispatch<AppState, unknown, never>
+) => {
   const txSortedByDescendingNonce = transactionData.sort(
+    // @ts-expect-error TypeScript prevents this subtraction since `n1` or `n2`
+    // may be null, but subtracting null and a number is possible.
     ({ nonce: n1 }, { nonce: n2 }) => n2 - n1
   );
   const [latestTx] = txSortedByDescendingNonce;
+  // @ts-expect-error `ZerionTransaction` doesn't have a `network` field, but
+  // the undefined network is defaulted to mainnet later.
   const { address_from, network, nonce } = latestTx;
```

Comment:
I wasn't able to find documentation for `ZerionTransaction` having a `network` field, and printing it out doesn't include one. What seems to happen is that `network` becomes undefined, but then is defaulted to mainnet by the function being called.

Reply:
yes this is correct! we treat undefined `network` as mainnet, and zerion does not provide this.